### ISSUE TITLE
[Numpy] np.where fails for scalar

### ISSUE THREAD

**Issue by ZheyuYe** [apache/incubator-mxnet#18416]
As title described,  a fatal error occurs when this operation encounters a scalar.  Same operation `nd.where` does not support scalar but only `nd.ndarray`.
```
import mxnet as mx
mx.npx.set_np()
condition = mx.np.array([1, 0, 1, 0, 1])
x = mx.np.ones_like(condition)
y = mx.np.zeros_like(condition)
mx.np.where(condition, x, y)
mx.np.where(condition, 1, 0)
```
Output:
```
array([1., 0., 1., 0., 1.])
array([0., 0., 0., 0., 0.])
```
A worse result emerged as 
```
mx.np.where(condition, 2, 103)
<<< array([103.,   0., 103.,   0., 103.])
```

**Comment by sxjscience** [comment]
I can confirm that there is a bug here.

**Comment by sxjscience** [comment]
@yzhliu I tagged it as 2.0 since it gives wrong result for `np.where`.

**Comment by Tommliu** [comment]
I think this issue is because of the wrong param assignment in FFI

>param.x = args[1].operator double();
  param.x = args[2].operator double(); 

I will add a quick fix to that.

**Comment by yzhliu** [comment]
should be fixed by #18451
